python/mlcroissant/mlcroissant/_src/structure_graph/nodes/record_set.py

```diff
@@ -108,30 +114,58 @@ def from_jsonld(cls, ctx: Context, record_set: Json) -> RecordSet:
             key=key,
             fields=fields,
             name=record_set_name,
-            uuid=uuid_from_jsonld(record_set),
+            id=uuid_from_jsonld(record_set),
         )
 
-    def check_joins_in_fields(self, fields: list[Field]):
+    def check_joins_in_fields(self):
         """Checks that all joins are declared when they are consumed."""
         joins: set[tuple[str, str]] = set()
         sources: set[str] = set()
-        for field in fields:
-            source_uid = field.source.uid
-            references_uid = field.references.uid
-            if source_uid:
-                # source_uid is used as a source.
-                sources.add(get_parent_uid(source_uid))
-            if source_uid and references_uid:
+        for field in self.fields:
+            source_uuid = field.source.uuid
+            references_uuid = field.references.uuid
+            if source_uuid:
+                # source_uuid is used as a source.
+                sources.add(get_parent_uuid(self.ctx, source_uuid))
+            if source_uuid and references_uuid:
                 # A join happens because the user specified `source` and `references`.
-                joins.add((get_parent_uid(source_uid), get_parent_uid(references_uid)))
-                joins.add((get_parent_uid(references_uid), get_parent_uid(source_uid)))
+                joins.add(
+                    (
+                        get_parent_uuid(self.ctx, source_uuid),
+                        get_parent_uuid(self.ctx, references_uuid),
+                    )
+                )
+                joins.add(
+                    (
+                        get_parent_uuid(self.ctx, references_uuid),
+                        get_parent_uuid(self.ctx, source_uuid),
+                    )
+                )
         for combination in itertools.combinations(sources, 2):
             if combination not in joins:
                 # Sort for reproducibility.
-                ordered_combination = tuple(sorted(combination))
+                ordered_combination = tuple(
+                    sorted([uuid_to_jsonld(source) for source in combination])
+                )
                 self.add_error(
                     f"You try to use the sources with names {ordered_combination} as"
                     " sources, but you didn't declare a join between them. Use"
                     " `ml:references` to declare a join. Please, refer to the"
                     " documentation for more information."
                 )
+
+
+def get_parent_uuid(ctx: Context, uuid: str) -> str:
+    """Retrieves the UID of the parent, e.g. `file/column` -> `file`."""
+    # TODO: rename this function and have better error message.
+    node = ctx.node_by_uuid(uuid)
+    if node is None:
+        raise ValidationError(
```

Comment:
Do not raise, but populate `ctx.issues.add_error`.

Reply:
Done.